atomic_reactor/plugins/post_group_manifests.py

```diff
@@ -0,0 +1,121 @@
+            if registry_conf.get('version') == 'v1':
+                continue
+
+            if not registry.startswith('http://') and not registry.startswith('https://'):
+                registry = 'https://' + registry
+
+            registry_noschema = urlparse(registry).netloc
+            for push_conf_registry in self.workflow.push_conf.docker_registries:
+                if push_conf_registry.uri == registry_noschema:
+                    break
+            else:
+                continue
+
+            insecure = push_conf_registry.insecure
+            auth = None
+            secret_path = registry_conf.get('secret')
+            if secret_path:
+                self.log.debug("registry %s secret %s", registry_noschema, secret_path)
+                dockercfg = Dockercfg(secret_path).get_credentials(registry_noschema)
+                try:
+                    username = dockercfg['username']
+                    password = dockercfg['password']
+                except KeyError:
+                    self.log.error("credentials for registry %s not found in %s",
+                                   registry_noschema, secret_path)
+                else:
+                    self.log.debug("found user %s for registry %s", username, registry_noschema)
+                    auth = requests.auth.HTTPBasicAuth(username, password)
+
+            digest = worker_digests[0]['digest']
+            tag = worker_digests[0]['tag']
+            repo = tag.split(':')[0]
+
+            url = '{}/v2/{}/manifests/{}'.format(registry, repo, digest)
+            response = requests.get(url, verify=not insecure, auth=auth)
+
+            image_manifest = json.loads(response.text)
+
+            if image_manifest['schemaVersion'] == '1':
+                msg = 'invalid schema from {}'.format(url)
+                raise PluginFailedException(msg)
+
+            for image in self.workflow.tag_conf.images:
+                image_tag = image.to_str(registry=False)
+                url = '{}/v2/{}/manifests/{}'.format(registry, repo, image_tag)
+                response = requests.put(url, data=json.dumps(image_manifest),
+                                        verify=not insecure, auth=auth)
+            worker_manifest.append(image_manifest)
+
+        return worker_manifest
+
+    def run(self):
+        if self.group:
+            raise NotImplementedError('group=True is not supported in group_manifest')
+        grouped_manifests = []
+
+        valid = False
+        all_annotations = self.workflow.build_result.annotations
+        for platform in all_annotations['worker-builds']:
+            if self.goarch.get(platform, platform) == 'amd64':
+                valid = True
+                grouped_manifests = self.get_worker_manifest(all_annotations['worker-builds'][platform])  # noqa
+                break
+
+        if valid:
+            return grouped_manifests
+        else:
+            raise PluginFailedException('failed to find an x86_64 platform')
```

Comment:
The exception raised here will be wrapped in a `PluginFailedException`. Let's use something else. Seems like a good case for `RuntimeError` or `ValueError`.